core/services/log/orm.go

```diff
@@ -54,44 +76,164 @@ func (o *orm) WasBroadcastConsumed(blockHash common.Hash, logIndex uint, jobID i
 	return consumed, err
 }
 
-func (o *orm) FindConsumedLogs(fromBlockNum int64, toBlockNum int64) ([]LogBroadcast, error) {
+func (o *orm) FindBroadcasts(fromBlockNum int64, toBlockNum int64) ([]LogBroadcast, error) {
 	var broadcasts []LogBroadcast
 	query := `
-		SELECT block_hash, log_index, job_id FROM log_broadcasts
+		SELECT block_hash, consumed, log_index, job_id FROM log_broadcasts
 		WHERE block_number >= $1
 		AND block_number <= $2
 		AND evm_chain_id = $3
-		AND consumed = true
 	`
 	err := o.db.Select(&broadcasts, query, fromBlockNum, toBlockNum, o.evmChainID)
 	if err != nil {
-		return make([]LogBroadcast, 0), err
+		return nil, errors.Wrap(err, "failed to find log broadcasts")
 	}
 	return broadcasts, err
 }
 
+func (o *orm) CreateBroadcast(blockHash common.Hash, blockNumber uint64, logIndex uint, jobID int32, qopts ...postgres.QOpt) error {
+	q := postgres.NewQ(o.db, qopts...)
+	res, err := q.Exec(`
+        INSERT INTO log_broadcasts (block_hash, block_number, log_index, job_id, created_at, updated_at, consumed, evm_chain_id)
+		VALUES ($1, $2, $3, $4, NOW(), NOW(), false, $5)
+    `, blockHash, blockNumber, logIndex, jobID, o.evmChainID)
+	if err != nil {
+		return errors.Wrap(err, "failed to create log broadcast")
+	}
+	rowsAffected, err := res.RowsAffected()
+	if err != nil {
+		return err
+	} else if rowsAffected == 0 {
+		return sql.ErrNoRows
```

Comment:
This error feels misplaced. TBH it seems impossible, there's probably no need to handle this.